### PR TITLE
Add authentication feature

### PR DESCRIPTION
**Pull request by HAYATO-Kz**
- Add authentication

### CODE REVIEW

File: components/Input.tsx [hunk 1]
```diff
@@ -1,36 +1,68 @@
 import React from 'react';
 import styled from 'styled-components/native';
-import {TextInputProps} from 'react-native';
+import { TextInputProps } from 'react-native';
 
 interface InputProps extends TextInputProps {
   /**
    * Inout label
```
Comment (NokKbl) [inline components/Input.tsx]:
```suggestion
   * Input label
```

File: screens/Login.tsx [hunk 2]
```diff
@@ -15,22 +16,35 @@ interface LoginFormValueProps {
...
    */
-  const handleLogin = (
+  const handleLogin = async (
     values: LoginFormValueProps,
     actions: FormikHelpers<LoginFormValueProps>,
   ) => {
-    console.log(JSON.stringify(values, null, 2));
+    setLoading(true);
+    const response = await axios
+      .post('http://localhost:3000/signIn', values)
```
Comment (NokKbl) [inline screens/Login.tsx:36]: Note: Don't forget to change this line later on

File: screens/Login.tsx [hunk 3]
```diff
@@ -15,22 +16,35 @@ interface LoginFormValueProps {
...
+            .catch((error) => {
+              const {message} = error.response.data
+        Alert.alert(message)
+      });
+    if (response) {
+      const { authToken } = response.data;
+      await AsyncStorage.setItem('authToken', 'tokenJa');
```
Comment (NokKbl) [inline screens/Login.tsx]: Could you change the name of `tokenJa` to be something meaningful

File: screens/Login.tsx [hunk 4]
```diff
@@ -39,15 +53,19 @@ const Login = (props: LoginProps) => {
       <Icon
         type="Entypo"
         name="chevron-left"
-        style={{color: 'black'}}
+        style={{ color: 'black' }}
         onPress={() => {
           navigation.navigate('Home');
         }}></Icon>
       <HeaderText>Log in</HeaderText>
```
Comment (NokKbl) [inline screens/Login.tsx:60]: When you will change to Thai version?

File: screens/Login.tsx [hunk 5]
```diff
@@ -15,22 +16,35 @@ interface LoginFormValueProps {
   password: string;
 }
 
-const INITIALLOGINFORM: LoginFormValueProps = {email: '', password: ''};
+const INITIALLOGINFORM: LoginFormValueProps = { email: '', password: '' };
```
Comment (NokKbl) [inline screens/Login.tsx]:
```suggestion
const INITIAL_LOGIN_FORM: LoginFormValueProps = { email: '', password: '' };
```

File: screens/Profile.tsx [hunk 6]
```diff
@@ -1,12 +1,28 @@
...
+
+  const handleSignOut = async () => {
+    await AsyncStorage.removeItem('authToken');
+    navigation.navigate('Home');
+  };
+
+  return (
+    <View>
+      <SafeAreaView />
+      <Text>Profile</Text>
```
Comment (NokKbl) [inline screens/Profile.tsx:22]: Thai word?

File: screens/Register.tsx [hunk 7]
```diff
@@ -16,19 +17,21 @@ interface RegisterFormValueProps {
   password: string;
 }
 
-const INITIALRegisterFORM: RegisterFormValueProps = {
+const INITIALREGISTERFORM: RegisterFormValueProps = {
```
Comment (NokKbl) [inline screens/Register.tsx]:
```suggestion
const INITIAL_REGISTER_FORM: RegisterFormValueProps = {
```

File: screens/Register.tsx [hunk 8]
```diff
@@ -45,23 +48,42 @@ const registerValidator = (values: RegisterFormValueProps) => {
   if (!values.username) {
     errors.username = 'Required';
   }
+
+  //  No error detected!!!!
```
Comment (NokKbl) [inline screens/Register.tsx]:
```suggestion
  // No error detected
```

File: screens/Register.tsx [hunk 9]
```diff
@@ -45,23 +48,42 @@ const registerValidator = (values: RegisterFormValueProps) => {
...
-  const handleSignUp = (
+  const handleSignUp = async (
     values: RegisterFormValueProps,
     actions: FormikHelpers<RegisterFormValueProps>,
   ) => {
-    console.log(JSON.stringify(values, null, 2));
-    actions.setSubmitting(false);
+    setLoading(true);
+    const response = await axios
+      .post('http://localhost:3000/signUp', values)
```
Comment (NokKbl) [inline screens/Register.tsx:75]: Note: Don't forget to change this later on

File: screens/Register.tsx [hunk 10]
```diff
@@ -70,23 +92,36 @@ const Register = (props: RegisterProps) => {
...
-        validate={registerValidator}>
-        {({handleChange, handleSubmit, values}) => (
+        validate={registerValidator}
+        validateOnChange={false}>
+        {({ handleChange, handleSubmit, errors, values }) => (
           <View>
-            <Input label="Username" onChangeText={handleChange('username')} />
-            <Input label="Email" onChangeText={handleChange('email')} />
+            <Input
+              label="Username"
```
Comment (NokKbl) [inline screens/Register.tsx:108]: Thai word?

**Comment by NokKbl** [comment]
Reviewed

File: screens/Login.tsx (hunk 4 above)
Comment (HAYATO-Kz) [reply screens/Login.tsx:60]: In refactor phase

File: screens/Profile.tsx (hunk 6 above)
Comment (HAYATO-Kz) [reply screens/Profile.tsx:22]: In refactor phase

File: screens/Register.tsx (hunk 10 above)
Comment (HAYATO-Kz) [reply screens/Register.tsx:108]: In refactor phase

**Review by NokKbl — APPROVED** [review]
LGTM

**Pull request merged**